velox/exec/TableScan.cpp

```diff
@@ -70,8 +72,12 @@ RowVectorPtr TableScan::getOutput() {
       const auto& connectorSplit = split.connectorSplit;
       needNewSplit_ = false;
 
-      if (!connector_) {
-        connector_ = connector::getConnector(connectorSplit->connectorId);
+      VELOX_CHECK(
+          connector_->connectorId() == connectorSplit->connectorId,
+          "Got splits with different connector IDs");
+
+      connector_ = connector::getConnector(connectorSplit->connectorId);
```

Comment:
Do we need this line?

Reply:
The constructor already initializes this right?

Reply:
Nice catch. We don't need this. Let me remove.